src/SFA.DAS.LevyTransferMatching.Web/Controllers/OpportunitiesController.cs

```diff
@@ -1,27 +1,72 @@
 ﻿using Microsoft.AspNetCore.Mvc;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Authorization;
-using SFA.DAS.LevyTransferMatching.Web.Attributes;
 using SFA.DAS.LevyTransferMatching.Web.Orchestrators;
+using SFA.DAS.LevyTransferMatching.Web.Models.Opportunities;
+using SFA.DAS.Authorization.Mvc.Attributes;
+using SFA.DAS.LevyTransferMatching.Web.Authentication;
+using SFA.DAS.LevyTransferMatching.Web.Attributes;
+using SFA.DAS.Authorization.EmployerUserRoles.Options;
 
 namespace SFA.DAS.LevyTransferMatching.Web.Controllers
 {
-    [AllowAnonymous]
-    [Route("opportunities")]
     [HideAccountNavigation(true)]
     public class OpportunitiesController : Controller
     {
-        private readonly IOpportunitiesOrchestrator _searchFundingOrchestrator;
+        private readonly IAuthenticationService _authenticationService;
+        private readonly IOpportunitiesOrchestrator _opportunitiesOrchestrator;
 
-        public OpportunitiesController(IOpportunitiesOrchestrator searchFundingOrchestrator)
+        public OpportunitiesController(IAuthenticationService authenticationService, IOpportunitiesOrchestrator searchFundingOrchestrator)
         {
-            _searchFundingOrchestrator = searchFundingOrchestrator;
+            _authenticationService = authenticationService;
+            _opportunitiesOrchestrator = searchFundingOrchestrator;
         }
 
         public async Task<IActionResult> Index()
         {
-            var viewModel = await _searchFundingOrchestrator.GetIndexViewModel();
+            var viewModel = await _opportunitiesOrchestrator.GetIndexViewModel();
             return View(viewModel);
         }
+
+        [Route("opportunities/{encodedPledgeId}")]
+        public async Task<IActionResult> Detail(DetailRequest detailRequest)
+        {
+            var viewModel = await _opportunitiesOrchestrator.GetDetailViewModel((int)detailRequest.PledgeId);
```

Comment:
Why the cast to Int? Does the request have to use a long? Is it to do with it being decoded? If so, we should fix the decoding to decode to an int where appropriate.

Reply:
Yes. I believe the framework used under the hood (Hashids) packs/unpacks `long`s - I guess in theory, if the hashid was longer than an `int`, you _could_ run into issues.

I tried creating the property as an `int`, but it failed to decode. It needs a `long`.

I can take a look at altering the shared package, if you like?

Reply:
I don't think it's a shared package thing, I can take a look this afternoon